abcclassroom/tests/test_clone.py

```diff
@@ -32,6 +35,37 @@ def test_files(default_config, tmp_path):
             Path(assignment_path, f).touch()
 
 
+# TODO: Test that the correct message is returned when no course_materials dir
+# exists in the config
+
+
+def test_roster_missing(sample_course_structure, tmp_path):
+    """Test that when the roster is missing things fail gracefully."""
+    # config = sample_course_structure
+    assignment_name = "test_assignment"
+
+    with pytest.raises(FileNotFoundError, match="Cannot find roster file"):
+        abcclone.clone_repos(assignment_name, skip_existing=False)
+
+
+# TODO: Test what happens when the roster exists but has the incorrect format
+def test_roster_wrong_format(sample_course_structure, tmp_path):
+    """Test that when the roster is missing things fail gracefully."""
+
+    course_name, config = sample_course_structure
+    assignment_name = "test_assignment"
+    # config = cf.get_config()
+    path_to_course = Path(config["course_directory"])
+    # Mess up the roster file
+    path_to_course.joinpath("classroom_roster.csv").touch()
```

Comment:
here i should write some header info that is incorrect because otherwise it's not even hitting the for loops?

we may want another test that there is something in the file???

Reply:
maybe missing headers